### PR TITLE
[2.8_stage] Update the generator docs

### PR DESCRIPTION
**Pull request by dhaiducek**
- Add `generate*Placement` fields
- Add `policySetDefaults`

ref: https://issues.redhat.com/browse/ACM-3011

### CODE REVIEW

File: governance/policy_generator.adoc [hunk 1]
```diff
@@ -535,7 +535,7 @@ roleRef:
 [#policy-gen-yaml-table]
 == Policy generator configuration reference table
 
-Note that all the fields in the `policyDefaults` section except for `namespace` can be overridden per policy.
+Note that all the fields in the `policyDefaults` section except for `namespace` can be overridden per policy, and all the fields in the `policySetDefaults` section can be overridded per policy set.
```
Comment (swopebe) [inline governance/policy_generator.adoc]: changing to "for each"

Comment (swopebe) [reply governance/policy_generator.adoc]: overridded? overridden?

File: governance/policy_generator.adoc [hunk 2]
```diff
@@ -689,6 +689,10 @@ Note that all the fields in the `policyDefaults` section except for `namespace`
 | Optional
 | Array of policy sets that the policy joins. Policy set details can be defined in the `policySets` section. When a policy is part of a policy set, a placement binding is not generated for the policy since one is generated for the set. Set `policies[].generatePlacementWhenInSet` or `policyDefaults.generatePlacementWhenInSet` to override `policyDefaults.policySets`.
 
+| `policyDefaults.generatePolicyPlacement`
+| Optional
+| Whether to generate placement manifests for policies. Setting to "false" will skip placement manifest generation, even if a placement is specified. This defaults to "true".
```
Comment (swopebe) [inline governance/policy_generator.adoc]: eliminating quotes and using `codeblock`

File: governance/policy_generator.adoc [hunk 3]
```diff
@@ -725,6 +729,18 @@ Note that all the fields in the `policyDefaults` section except for `namespace`
 | Optional
 | Specify a placement rule by defining a cluster selector in the following format, `key:value`. See `placementRulePath` to specify an existing file.
 
+| *`policySetDefaults`*
```
Comment (swopebe) [inline governance/policy_generator.adoc]: removing bold.

File: governance/policy_generator.adoc [hunk 4]
```diff
@@ -725,6 +729,18 @@ Note that all the fields in the `policyDefaults` section except for `namespace`
...
+
+| `policySetDefaults.placement`
+| Optional
+| The placement configuration for the policies. This defaults to a placement configuration that matches all clusters. (See `policyDefaults.placement` for description of this field.)
+
+| `policySetDefaults.generatePolicySetPlacement`
+| Optional
+| Whether to generate placement manifests for policy sets. Setting to "false" will skip placement manifest generation, even if a placement is specified. This defaults to "true".
+
 | *`policies`*
```
Comment (swopebe) [inline governance/policy_generator.adoc]: same

File: governance/policy_generator.adoc [hunk 5]
```diff
@@ -747,7 +763,7 @@ Note that all the fields in the `policyDefaults` section except for `namespace`
 
 | *`policySets`*
```
Comment (swopebe) [inline governance/policy_generator.adoc]: same

**Review by swopebe — APPROVED** [review]
Everything mentioned is resolved

**Comment by dhaiducek** [comment]
@swopebe Thanks for the quick review, but I opened this PR proactively for `2.8`. It wasn't meant to be merged into `2.7`--sorry if that was confusing/unclear. Let me know how I can help to roll it back.

**Comment by swopebe** [comment]
@dhaiducek , we don't have 2.8_stage, that comes much later every release, post GA. That is why we really need issues opened, but 2.8 changes are not possible right now. (We usually announce when 2.8 is open, that is why I assumed this was for 2.7, yes, we need to remove this before GA this week.)

**Comment by dhaiducek** [comment]
Sounds good, sorry for the confusion! This PR has already been reverted in:
- https://github.com/stolostron/rhacm-docs/pull/4591

I'll open a new PR once `2.8` opens. 🙂

**Pull request merged**